sklearn/utils/validation.py

```diff
@@ -449,6 +449,17 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
     # DataFrame), and store them. If not, store None.
     dtypes_orig = None
     if hasattr(array, "dtypes") and hasattr(array.dtypes, '__array__'):
+        # throw warning if pandas dataframe is sparse
+        try:
```

Comment:
Instead of this try except:

```py
from contextlib import suppress

...

with supress(ImportError):
    from pandas.api.types import is_sparse
	... 
```